src/java/org/apache/cassandra/auth/DataResource.java

```diff
@@ -249,7 +250,8 @@ public boolean exists()
                 return true;
             case KEYSPACE:
             case ALL_TABLES:
-                return Schema.instance.getKeyspaces().contains(keyspace);
+                return SchemaConstants.isVirtualSystemKeyspace(keyspace) ||
+                       Schema.instance.getKeyspaces().contains(keyspace);
```

Comment:
Nit: I think it may be useful to add a comment here on why we are including the `isVirtualSystemKeyspace`

Reply:
Not sure about that, since the commit message will stick on that change linking the JIRA ticket which explains it all.